### PR TITLE
Add Eureka provider

### PR DESCRIPTION
**Pull request by Juliens**
- [x] "Provide" Eureka Applications
- [x] auto-refresh (with interval in configuration)
- [x] overload configuration in Eureka Metadata (only weight)
- [x] HTTPS ?
- [x] Update documentation
- [x] tests unit

### CODE REVIEW

File: provider/eureka.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+// Provide allows the provider to provide configurations to traefik
+// using the given configuration channel.
+func (provider *Eureka) Provide(configurationChan chan<- types.ConfigMessage, pool *safe.Pool, _ []types.Constraint) error {
+
+	configuration, err := provider.buildConfiguration()
+	if err != nil {
+		log.Errorf("Failed to build configuration for Eureka, error: %s", err)
+		return nil
```
Comment (emilevauge) [inline provider/eureka.go]: You should return the error.

**Comment by emilevauge** [comment]
@Juliens thanks a lot for this great contribution :)
I have few comments:
- I think it would be great to use backoff retries in your provide function, in case of failing connection (like what is done here https://github.com/containous/traefik/blob/master/provider/docker.go#L142)
- do you think it's possible to add an integration test with this new provider (using https://hub.docker.com/r/netflixoss/eureka)?
- I did not find anyway to do this but do you know if Eureka supports long polling to get push notifications at each change?

**Comment by Russell-IO** [comment]
LGTM

**Comment by emilevauge** [comment]
Hi, traefik v1.1 developments will be frozen this week, on Thursday 29th. If you want your PR to be part of this release, please update your PR as soon as possible :)

**Comment by emilevauge** [comment]
Ping?

**Comment by errm** [comment]
Hi @Juliens it would be great to get this into the 1.2 release... once 1.1 is out shortly it would be great to get this merged in quickly... Just need to resolve the conflicts so we are ready to merge...

**Comment by emilevauge** [comment]
@Juliens golint has been updated yesterday, which lead to travis errors while validating.
@errm fixed that on master, could you rebase again?

**Pull request merged**